diff --git a/tests/test_public_key.py b/tests/test_public_key.py
index 5cb88fb..1a2c520 100644
--- a/tests/test_public_key.py
+++ b/tests/test_public_key.py
@@ -1004,13 +1004,13 @@ class _TestPublicKey(TempDirTestCase):
             ('Invalid PEM PKCS#8 PBES2 encryption parameters',
              b'-----BEGIN ENCRYPTED PRIVATE KEY-----\n' +
              binascii.b2a_base64(der_encode(
-                 ((_ES2, ((_ES2_PBKDF2, (b'', 0)),
+                 ((_ES2, ((_ES2_PBKDF2, (b'', 1)),
                           (_ES2_AES128, None))), b''))) +
              b'-----END ENCRYPTED PRIVATE KEY-----'),
             ('Invalid length PEM PKCS#8 PBES2 IV',
              b'-----BEGIN ENCRYPTED PRIVATE KEY-----\n' +
              binascii.b2a_base64(der_encode(
-                 ((_ES2, ((_ES2_PBKDF2, (b'', 0)),
+                 ((_ES2, ((_ES2_PBKDF2, (b'', 1)),
                           (_ES2_AES128, b''))), b''))) +
              b'-----END ENCRYPTED PRIVATE KEY-----'),
             ('Invalid OpenSSH cipher',
